x-pack/plugins/lens/public/datatable_visualization/components/table_basic.tsx

```diff
@@ -286,6 +287,39 @@ export const DatatableComponent = (props: DatatableRenderProps) => {
     [onEditAction, sortBy, sortDirection]
   );
 
+  const renderSummaryRow = useMemo(() => {
+    const columnsWithSummary = columnConfig.columns
+      .map((config) => ({
+        columnId: config.columnId,
+        summaryRowValue: config.summaryRowValue,
+        ...getFinalSummaryConfiguration(config.columnId, config, firstTable),
+      }))
+      .filter(({ summaryRow }) => summaryRow !== 'none');
```

Comment:
If the only column with a summary row is hidden, it will show a weird empty row (note the thin gray line below the thick black one):
<img width="1629" alt="Screenshot 2021-06-08 at 14 05 02" src="https://user-images.githubusercontent.com/1508364/121182210-24398900-c863-11eb-8275-b61fe65e681b.png">

Should be easy to fix take into account the hidden flag and filter `columnsWithSummary`